### PR TITLE
Replace release label validation script

### PR DESCRIPTION
**Pull request by MarekMichali**
**Description**

Replace the existing bash label validation script with a python script. This new script will only be used during release. Validation at the PR level will still be done by the old bash script.

Changes proposed in this pull request:

**Related issue(s)**

See also #98

### CODE REVIEW

File: scripts/python/release_label_validator.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+invalid_prs = []
+for pr in prs_since_last_release:
+    labels = [label['name'] for label in pr['labels']]
+    common_labels = set(labels).intersection(label_pool)
+    if len(common_labels) != 1:
+        invalid_prs.append(pr['html_url'])
+    else:
+        valid_prs.append(pr['html_url'])
+
+print("\nThese PRs have exactly one label from the pool:")
```
Comment (jaroslaw-pieszka) [inline scripts/python/release_label_validator.py:43]: `pool` term is kind of vague here.

File: scripts/python/release_label_validator.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+for pr_url in valid_prs:
+    print(f"PR: {pr_url}")
+
+
+if invalid_prs:
+    print("\nThese PRs don't have exactly one label from the pool:")
+    for pr_url in invalid_prs:
```
Comment (jaroslaw-pieszka) [inline scripts/python/release_label_validator.py:50]: We could join all references with a separator and print in separate line with newline at the end.

File: scripts/python/release_label_validator.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+latest_release = response.json()
+latest_release_date = latest_release['created_at']
+
+response = requests.get(f'https://api.github.com/repos/{repo}/pulls?state=closed&sort=updated&direction=desc', headers={'Authorization': f'token {token}'})
+response.raise_for_status()
+all_closed_prs = response.json()
+
+prs_since_last_release = [
+    pr for pr in all_closed_prs
+    if pr['merged_at'] is not None and pr['merged_at'] > latest_release_date
```
Comment (jaroslaw-pieszka) [inline scripts/python/release_label_validator.py:30]: How about these merged after the release was created but not taken by release.
I mean is it possible that merged_at > latest_release_date but the point in time when we created tag and cut off some PRs to be released is after latest_release_date.

File: scripts/python/release_label_validator.py (hunk 1 above)
Comment (MarekMichali) [reply scripts/python/release_label_validator.py]: Changed to `required labels`

**Review by jaroslaw-pieszka — APPROVED** [review]

**Pull request merged**